src/test/java/com/flagsmith/FlagsmithApiWrapperTest.java

```diff
@@ -213,106 +126,38 @@ public void identifyUserWithTraits_success() throws JsonProcessingException {
-    user.setIdentifier("user-w-traits");
-    interceptor.addRule()
-        .post(BASE_URL + "/identities/")
-        .respond(mapper.writeValueAsString(newFlagsAndTraits()), MEDIATYPE_JSON);
-    defaultConfig.flagsmithFlagDefaults.setDefaultFeatureFlags(new HashSet<String>() {{
-      add("default-flag");
-    }});
-
-    // Act
-    final FlagsAndTraits actualFeatureFlags = sut.identifyUserWithTraits(user, traits, true);
+    final Flags actualFeatureFlags = sut.identifyUserWithTraits("user-w-traits", traits, false);
 
     // Assert
-    final FlagsAndTraits expectedFlags = newFlagsAndTraits();
-    expectedFlags.getFlags().add(flag("default-flag", null, "FLAG", false, null));
-    assertEquals(expectedFlags, actualFeatureFlags);
-    assertEquals(expectedFlags, actualFeatureFlags);
+    assertEquals(newFlagsList(new ArrayList<>()), actualFeatureFlags);
     verify(flagsmithLogger, times(1)).info(anyString(), any(), any());
-    verify(flagsmithLogger, times(0)).httpError(any(), any(Response.class), anyBoolean());
-    verify(flagsmithLogger, times(0)).httpError(any(), any(IOException.class), anyBoolean());
-  }
-
-  @Test(groups = "unit")
-  public void postUserTraits_success() throws JsonProcessingException {
-    // Arrange
-    final FeatureUser user = new FeatureUser();
-    user.setIdentifier("username");
-    final Trait inputTrait = new Trait();
-    final Trait expectedTrait = new Trait();
-    expectedTrait.setValue("some-value");
-    interceptor.addRule()
-        .post(BASE_URL + "/traits/")
-        .respond(mapper.writeValueAsString(expectedTrait), MEDIATYPE_JSON);
-
-    // Act
-    final Trait actualTrait = sut.postUserTraits(user, inputTrait, true);
-
-    // Assert
-    assertEquals(expectedTrait, actualTrait);
-    assertEquals(user.getIdentifier(), inputTrait.getIdentity().getIdentifier());
-    verify(flagsmithLogger, times(1)).info(anyString(), any(), any(), any());
-    verify(flagsmithLogger, times(0)).httpError(any(), any(Response.class), anyBoolean());
-    verify(flagsmithLogger, times(0)).httpError(any(), any(IOException.class), anyBoolean());
-  }
-
-  @Test(groups = "unit")
-  public void postUserTraits_fail() {
-    // Arrange
-    final FeatureUser user = new FeatureUser();
-    user.setIdentifier("username");
-    interceptor.addRule()
-        .post(BASE_URL + "/traits/")
-        .respond(500, ResponseBody.create("error", MEDIATYPE_JSON));
-
-    // Act
-    final Trait actualTrait = sut.postUserTraits(user, new Trait(), false);
-
-    // Assert
-    assertNull(actualTrait);
-    verify(flagsmithLogger, times(1)).info(anyString(), any(), any(), any());
     verify(flagsmithLogger, times(1)).httpError(any(), any(Response.class), eq(false));
     verify(flagsmithLogger, times(0)).httpError(any(), any(IOException.class), anyBoolean());
   }
 
-  private FlagsAndTraits newFlagsAndTraits() {
-    final Feature feature = new Feature();
+  private FeatureStateModel getNewFlag() {
```

Comment:
May be we can add Api Response classes to properly address this concern, like FlagsApiResponse etc. It would essentially extend FeatureStateModel anyways.

Reply:
I think that sounds like a good idea, yep.

Reply:
@fzia are we ok to make these changes?